mirgecom/boundary.py

```diff
@@ -653,7 +677,790 @@ def isothermal_noslip_state(self, discr, btag, gas_model, state_minus, **kwargs)
+
+        num_specified = sum(x is not None for x in [free_stream_temperature,
+                                                    free_stream_pressure,
+                                                    free_stream_density])
+        if num_specified != 2:
+            raise ValueError("Two of [free_stream_pressure, free_stream_temperature,"
+                             " free_stream_density], must be provided.")
+
+        PrescribedFluidBoundary.__init__(
+            self, boundary_state_func=self.farfield_state
+        )
+
+    def farfield_state(self, discr, btag, gas_model, state_minus, **kwargs):
+        """Get the exterior solution on the boundary."""
+        free_stream_mass_fractions = (0.*state_minus.species_mass_fractions
+                                      + self._species_mass_fractions)
+
+        if self._temperature is not None:
+            free_stream_temperature = 0.*state_minus.temperature + self._temperature
+        if self._pressure is not None:
+            free_stream_pressure = 0.*state_minus.pressure + self._pressure
+        if self._density is not None:
+            free_stream_density = 0.*state_minus.cv.mass + self._density
+
+        if self._temperature is None:
+            free_stream_temperature = \
+                (0.*state_minus.temperature
+                 + (self._pressure / (gas_model.eos.gas_const() * self._density)))
+        if self._pressure is None:
+            free_stream_pressure = \
+                gas_model.eos.gas_const()*self._density*self._temperature
+        if self._density is None:
+            free_stream_density = gas_model.eos.get_density(
+                pressure=free_stream_pressure, temperature=free_stream_temperature,
+                species_mass_fractions=free_stream_mass_fractions)
+
+        free_stream_velocity = 0.*state_minus.velocity + self._velocity
+        free_stream_internal_energy = gas_model.eos.get_internal_energy(
+            temperature=free_stream_temperature,
+            species_mass_fractions=free_stream_mass_fractions)
+
+        free_stream_total_energy = \
+            free_stream_density*(free_stream_internal_energy
+                                 + .5*np.dot(free_stream_velocity,
+                                             free_stream_velocity))
+        free_stream_spec_mass = free_stream_density * free_stream_mass_fractions
+
+        cv_infinity = make_conserved(
+            state_minus.dim, mass=free_stream_density,
+            energy=free_stream_total_energy,
+            momentum=free_stream_density*free_stream_velocity,
+            species_mass=free_stream_spec_mass
+        )
+
+        return make_fluid_state(cv=cv_infinity, gas_model=gas_model,
+                                temperature_seed=free_stream_temperature)
+
+    def temperature_bc(self, state_minus, **kwargs):
+        """Return farfield temperature for use in grad(temperature)."""
+        return 0*state_minus.temperature + self._temperature
```

Comment:
I am unsure this will work when "free_stream_temperature" is initially None sine here (https://github.com/illinois-ceesd/mirgecom/pull/678/files#diff-af7fe103044e2f2c2fb594902042f0c99e7c8221c6e94d0310bfed735ab96823R763) you are attributing the computed value to "free_stream_temperature" and not "self._temperature"

Reply:
We cannot set `self._temperature` down in this routine.   That one must be set in `__init__` if anywhere.   We may need some conditional processing in this routine to resolve the issue - but that's going to be tricky because we don't have access to the `gas_model` here.

Reply:
Can you just replace "self._temperature" here with "free_stream_temperature" ? It is the output from both "is None" and "is not None" checks.

Reply:
@tulioricci see what you think of the farfield now

Reply:
Looks good to me.